src/rust/ide/view/graph-editor/src/component/visualization/foreign/java_script/binding.rs

```diff
@@ -86,10 +94,57 @@ impl JsConsArgs {
         self.root.clone().into()
     }
 
+    /// Getter for the theming API that we expose to JS visualizations
+    pub fn theme(&self) -> JsTheme {
+        self.theme.clone()
+    }
+
     /// Helper method to emit an preprocessor change event from the visualisation.
     pub fn emit_preprocessor_change(&self, code:Option<String>, module:Option<String>){
         let closure             = &self.set_preprocessor;
         let preprocessor_config = PreprocessorConfiguration::from_options(code,module);
         (*closure)(preprocessor_config);
     }
 }
+
+/// The theming API that we expose to JS visualizations
+#[wasm_bindgen]
+#[derive(Clone, Debug)]
+pub struct JsTheme {
+    styles: StyleWatch
+}
+
```

Comment:
The whole JsTheme thing should be in a separate section. We also try to introduce first things that are used later, s this should go to earlier lines in the file.